File: src/org/newtonpolyhedron/solverprinters/MatrixInverseSolverPrinter.java
package org.newtonpolyhedron.solverprinters;

import java.io.PrintWriter;

import org.apache.commons.math3.fraction.BigFraction;
import org.apache.commons.math3.linear.FieldMatrix;
import org.newtonpolyhedron.entity.SolverPrinter;
import org.newtonpolyhedron.utils.MatrixUtils;

public class MatrixInverseSolverPrinter extends SolverPrinter <Void> {
	
	private final FieldMatrix <BigFraction>	baseMatrix;
	
	public MatrixInverseSolverPrinter(final FieldMatrix <BigFraction> baseMatrix, final PrintWriter output) {
		super(null, output);
		this.baseMatrix = baseMatrix;
	}
	
	@Override
	protected void solveFor(final Void nothing, final PrintWriter output) throws Exception {
		final FieldMatrix <BigFraction> inv = MatrixUtils.inverse(baseMatrix);
		output.println(subheader("Base matrix:"));
		output.println(MatrixUtils.toString(baseMatrix));
		output.println(subheader("Matrix inverse:"));
		output.println(MatrixUtils.toString(inv));
	}
}

File: src/org/newtonpolyhedron/solverprinters/MatrixInverseSolverPrinter.scala
package org.newtonpolyhedron.solverprinters
import java.io.PrintWriter

import org.apache.commons.math3.fraction.BigFraction
import org.apache.commons.math3.linear.FieldMatrix
import org.newtonpolyhedron.entity.BigFrac
import org.newtonpolyhedron.entity.Matrix
import org.newtonpolyhedron.entity.SolverPrinter

class MatrixInverseSolverPrinter(val baseMatrix: Matrix[BigFrac],
                                 output: PrintWriter)
    extends SolverPrinter[Void](null, output) {

  def this(baseMatrix: FieldMatrix[BigFraction],
           output: PrintWriter) =
    this(Matrix(baseMatrix.getData map (_ map (BigFrac(_)))), output)

  override def solveFor(nothing: Void,
                        output: PrintWriter) = {
    val inv = baseMatrix.inv
    output.println(subheader("Base matrix:"))
    output.println(baseMatrix)
    output.println(subheader("Matrix inverse:"))
    output.println(inv)
  }
}